src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -122,7 +220,45 @@ protected internal override void SetPinMode(int pinNumber, PinMode mode)
 
         protected internal override WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventType, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            if (!_pinNumberToEventHandler.TryGetValue(pinNumber, out LibGpiodDriverEventHandler eventHandler))
+            {
+                eventHandler = new LibGpiodDriverEventHandler(pinNumber, new CancellationTokenSource());
+                if (_pinNumberToEventHandler.TryAdd(pinNumber, eventHandler))
+                {
+                    eventHandler.PinHandle = SubscribeForEvent(pinNumber, eventType);
+                    eventHandler.Task = InitializeEventDetectionTask(eventHandler, eventHandler.CancellationTokenSource.Token);
+                }
+            }
+
+            CancellationTokenSource cancelWhenEventOccured = new CancellationTokenSource();
+            PinChangeEventHandler callback = (o, e) => {
+                cancelWhenEventOccured.Cancel();
+            };
+            if (eventType.HasFlag(PinEventTypes.Rising))
+            {
+                eventHandler.ValueRising += callback;
+            }
+            else if (eventType.HasFlag(PinEventTypes.Falling))
+            {
+                eventHandler.ValueFalling += callback;
+            }
+
+            WaitForEventResult(cancellationToken, eventHandler.CancellationTokenSource.Token, cancelWhenEventOccured.Token, eventType);
+            RemoveCallbackForPinValueChangedEvent(pinNumber, callback);
+
+            return new WaitForEventResult
+            {
+                TimedOut = !cancelWhenEventOccured.IsCancellationRequested,
+                EventType = eventType
+            };
+        }
+
+        private void WaitForEventResult(CancellationToken sourceToken, CancellationToken parentToken, CancellationToken eventOccuredToken,  PinEventTypes eventType)
+        {
+            while (!(sourceToken.IsCancellationRequested || parentToken.IsCancellationRequested || eventOccuredToken.IsCancellationRequested))
+            {
+                Thread.Sleep(1_000);
```

Comment:
why such a long sleep?

Reply:
😛 thought it is 1 millisecond, good catch